### PR TITLE
21952 Furnishings job - Implement stage three

### PR DESCRIPTION
**Pull request by leodube-aot**
*Issue #:* /bcgov/entity#21952

*Description of changes:*
- Implemented [stage 3 of furnishings job tech design](https://community.inkdrop.app/a27b7a79c8cdf7db0ab19be10b4fc2e8/lr3hwYXNS#stage-3).
- Added unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by argush3** [comment]
@kzdev420 @chenhongjing leaving this with you two as we have already implemented the equivalent functionality in stage 2.  Think it's low enough complexity at this time.

File: jobs/furnishings/src/furnishings/stage_processors/stage_three.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        grouping_identifier = Furnishing.get_next_grouping_identifier()
+
+        for batch_processing in batch_processings:
+            business = batch_processing.business
+            furnishing_name = (
+                Furnishing.FurnishingName.CORP_DISSOLVED_XPRO
+                if business.legal_type == Business.LegalTypes.EXTRA_PRO_A.value
+                else Furnishing.FurnishingName.CORP_DISSOLVED
+            )
+            new_furnishing = Furnishing(
```
Comment (chenhongjing) [inline jobs/furnishings/src/furnishings/stage_processors/stage_three.py:52]: Can you populate `business_name` for furnishing record as well?

**Review by chenhongjing — APPROVED** [review]
Looks good 😸 and pre-approve on the assumption that my comment will be resolved.

**Comment by argush3** [comment]
please review @kzdev420

**Review by kzdev420 — APPROVED** [review]

File: jobs/furnishings/src/furnishings/stage_processors/stage_three.py (hunk 1 above)
Comment (leodube-aot) [reply jobs/furnishings/src/furnishings/stage_processors/stage_three.py:52]: `Furnishing` doesn't have a `business_name` attribute. Do you mean populate the `business` relationship?

**Pull request merged**